### ISSUE TITLE
Open to making the file transport pluggable?

### ISSUE THREAD

**Issue by chmac** [onubo/react-native-logs#22]
Firstly, thanks for this logger. I've been searching for a while for something to use in React Native, and this seems to hit a pretty sweet spot!

I'm using expo, and I prefer to avoid the RNFS filesystem package if possible. I wrote a package [expo-fs](https://generouslabs.github.io/expo-fs/) to make isomorphic-git compatible with expo's filesystem. I was looking to write a custom transport for react-native-logs to do the same. Then looking at the code, I realise it's really only this one pieces that would need to be changed:

https://github.com/onubo/react-native-logs/blob/913eb01ac550970b659528ae04a80080bce85b4c/src/transports/rnFsFileAsync.ts#L44-L48

That got me to thinking, would you be open to a PR that makes this piece pluggable? I haven't thought much about how to achieve that yet, maybe turn it into a factory that takes an `appendFile()` function, then use that factory to generate the same transport as currently so it's fully backwards compatible, but easy enough to drop in something else in place.

Wanted to check in before I start hacking on anything, I can equally just copy that file into a separate package and write a `react-native-logs-expo-fs-transport` or similarly creatively named package! :-)

**Comment by chmac** [comment]
I forked the repo and started some experiments. On reflection, I don't think this approach makes sense. There's so much formatting and so on in the transport, that I guess it makes more sense to simply create my own transport in my app. I might also publish that as a package, not sure yet.

Expo doesn't have an appendFile, so I'll likely shard the files by date and so on as well.

**Comment by alessandro-bottamedi** [comment]
Thanks for the appreciation :), I don't really know how expo-fs works but i think the best idea is to create a new transport, it's the easiest way to do it.

If you create your transport with expo-fs then we can integrate it in the package

**Comment by chmac** [comment]
@alessandro-bottamedi I tried this approach this morning, just pushed the branch if you want to check it out. The approach worked locally. But the dependency on `expo-file-system` is invisible, which I'm not quite sure about. I guess it would make sense to publish transports which have dependencies as their own packages, so they're tracked separately.

I didn't end up using `expo-fs` as it's just another layer of abstraction which wasn't necessary in this case. If you want to add the expoFileSystem transport into core, the code I pushed might work.

I also added a prettier configuration and some other stuff, but I haven't pushed any of that. I'm leaning more and more towards my own app specific transport.
